tests/Microsoft.Graph.Test/Requests/Functional/SharePointTests.cs

```diff
@@ -99,14 +102,67 @@ public class SharePointTests : GraphTestBase
             }
         }
 
-        [Ignore] // Issue with this. Informed service API owner. Sharing token is not recognized.
+        /// <summary>
+        /// Tests the GetSiteByPath method added in GraphServiceSitesCollectionRequestBuilderExtension.cs
+        /// https://developer.microsoft.com/en-us/graph/docs/api-reference/v1.0/api/site_get
+        /// </summary>
+        /// Open question: how is a customer expected to get Site path. This part of the experience is unclear to me. 
         [TestMethod]
-        public async System.Threading.Tasks.Task SharePointAccessSiteByUrl()
+        public async Async.Task SharePointGetSiteWithPath()
         {
             try
             {
-                // 
-                Site site = graphClient.Shares[UrlToSharingToken("https://mod810997.sharepoint.com/sites/SMBverticals")].Site.Request().GetAsync().Result;
+                // Create the request to get the root site by using the root structural property. We don't generate 
+                // request builders for structural properties so we need to use HttpRequestMessage to make the request.
+                string requestUrlToGetSiteRootInfo = String.Format("{0}{1}", graphClient.Sites.Request().RequestUrl, "/root");
+                HttpRequestMessage hrm = new HttpRequestMessage(HttpMethod.Get, requestUrlToGetSiteRootInfo);
+
+                // Authenticate (add access token) to our HttpRequestMessage
+                await graphClient.AuthenticationProvider.AuthenticateRequestAsync(hrm);
+
+                HttpResponseMessage response = await graphClient.HttpProvider.SendAsync(hrm);
+
+                Site site;
+
+                // Get the Site.
+                if (response.IsSuccessStatusCode)
+                {
+                    // Deserialize Site object.
+                    var content = await response.Content.ReadAsStringAsync();
+                    site = graphClient.HttpProvider.Serializer.DeserializeObject<Site>(content);
+                }
+                else
+                    throw new ServiceException(
```

Comment:
Since we've seen this in a few places, maybe add this to a helper/util file? Something that converts the response object into an error?